---
title: Microbiota-derived Mitochondrial coenzymes 
author: 'Runhang'
date: '2021-01-21'

---

#### Systemic Regulation of Host Energy and Oogenesis by Microbiome-Derived Mitochondrial Coenzymes

Gnainsky et al., 2021 *Cell Reports* DOI:https://doi.org/10.1016/j.celrep.2020.108583 

### Systems: Female Drosophila melanogaster

### Rationale:

1. Lowered mitochondrial DNA and mRNA in the pre-mature ovary of germ-free (GF) flies but not in mature ovary, nor other regions such as gut and head. Supplementing *Acetobacter* restored egg production and the expression of mitochondrial genes. 

  * Thoughts: Basicly, they did some qPCR screenings as the first step. It points out a noval insight of mitochondrial gene and transcriptional responses to microbiome removal. The first piece of results stand alone can be put in *Plos one* but the authors continued to delineate more (such as the functions)!

2. From gene level to function level: they first measured **mitochondrial membrain potential (staining TMRE and YFP)** of various egg chamber stage (0-9), as an index of mitochondrial function; Subsequently, ATP anf ADP levels of whole body, mature oocytes, and head were quantified using qLC-MS. OK, now a systemical reduction in GF fly energy was observed, the next step is to confirm the corelation: **Is the reduction in mitochondrial function resulted in the repressed oogenesis in GF flies?**

3. Following the rationale of step2, the authors inhibited mitochondrial function by adding chemical inhibitor rotenome in the diet (permit bacterial growth). They found 25 uM of rotenone was sufficient to repress oogenesis, which mimicked the impact of bacterial removal. 

  * Thoughts: The oral feeding assay had limitations in that it resulted in a systemical inhibition in mitochondrial activity while it remains unknown that **whether tissue-specific downregulation of mitochondrial function in somatic follicle cells is sufficient to repress oogenesis**. 

4. Therefore, they did follicle-specific knockdown of two subunits of mitochondrial complex I by RNAi delivered by GAL4 system. The follicle-specific knockdown of either genes was sufficient to repress the mitochondrial membrane potential

5.  So far, they had understood that 1)reduction in oogenesis is a result of supressed mitochondrial function in follicle cells. 2) GF flies showed decreased mitochondrial membrain potenial and energy turnover. **But, what are the molecular mechanisms mediated by the bacteria?**

6. To start with, they probably started looking into literature and interrogated that why the bacterial symbiont - *Acetobacter spp.* can modulate mitochondrial activities? which pathway? Indeed, studies in Drosophila showed the symbiotic microbes are an integral part of the host nutrition including B vitamin provisioning (Wong et al., 2014; Sannino et al., 2018). Since vitamin B2 (RbF) is the precursor for two essential mitochondrial coenzymes (FAD and FMN), the authors therefore feed GF flies with dietary RbF. The results are remarkble: supplementation of RbF completely restored mitochondrial membrane potential, and patially restored the ATP levels in the ovary of GF flies. 

7. Metabolomics data (whole body and overy) support the GF's reduction in metabolites involved in main energy pathways including TCA cycle, Glycolysis, Pentose phosphate pathway. Higher levels of amino acids and lipids were observed in GF flies. 

8. The metabolites higher in GF flies are mostly known to be degraded by FAD-dependent enzymes. So the authors did "gain of function" assay: supplemented the RbF to GF again and measured the metabolites.  

9. Finally, they found the links between their hypothesis of *Acetobacter*-mediated mitochondrial function with another independent study that showed *Acetobacter*-mediated *Aldh* expression (Elgart et al., 2016).

### Closing Remarks
I assume this lab does not specialize in bacterial genomics. Otherwise, they should address the key question that has not been considered in this study: Does *Acetobacter spp* express genes such as *ribF* that encodes FMN and FAD in the flavin biosynthesis? However, they used a very smart way to skip dealing with bacterium genomics by using RbF supplementation. This is my only criticism about this nice paper.  

